src/Lean/Elab/Command.lean

```diff
@@ -503,6 +504,45 @@ def expandDeclId (declId : Syntax) (modifiers : Modifiers) : CommandElabM Expand
 
 end Elab.Command
 
+open Elab Command MonadRecDepth
+
+/--
+Lift an action in `CommandElabM` into `CoreM`, updating the traces and the environment.
+This does not preserve things like `open` and `namespace` declarations.
+-/
+def liftCommandElabM (k : CommandElabM α) : CoreM α := do
+  let (a, commandState) ←
+    k.run {
+      fileName := ← getFileName
+      fileMap := ← getFileMap
+      ref := ← getRef
+      tacticCache? := none
+    } |>.run {
+      env := ← getEnv
+      maxRecDepth := ← getMaxRecDepth
+      scopes := [{ header := "", opts := ← getOptions }]
+    }
+  modify fun coreState => { coreState with
+    traceState.traces := coreState.traceState.traces ++ commandState.traceState.traces
+    env := commandState.env
+  }
+  if let some err := commandState.messages.msgs.toArray.find? (·.severity matches .error) then
+    throwError err.data
+  pure a
+
+/--
+Evaluate any `set_option in` commands before the given `stx`, and pass the inner `stx` with the
+updated environment to the continuation `k`.
```

Comment:
This fixes indicative vs imperative:
```suggestion
Evaluates any `set_option in` commands before the given `stx`, and pass the inner `stx` with the
updated environment to the continuation `k`.
```
But I'd write it a bit differently, because `stx` isn't a named argument, and `withSetOptionIn` presents itself as a transformer of command elaborators, rather than as a monadic action that consumes syntax:
```suggestion
Given a command elaborator `elab`, returns a new command elaborator that first evaluates any local `set_option ... in ...` clauses and then invokes `elab` on what remains.
```
(while also renaming the argument to `elab`)